pkg/e2e/osd/dedicatedadmin.go

```diff
@@ -269,6 +271,37 @@ var _ = ginkgo.Describe(dedicatedAdminTestName, func() {
 			err := manageSubscriptions(ctx, namespaceList, h)
 			Expect(err).NotTo(HaveOccurred())
 		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// dedicated-admin user can patch consoles.operator.openshift.io CR
+		util.GinkgoIt("ded-admin user can patch consoles.operator.openshift.io CR", func(ctx context.Context) {
+			// Impersonate ded-admin
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "dummy-admin@redhat.com",
+				Groups: []string{
+					"dedicated-admin",
+				},
+			})
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+			}()
+
+			patchData := []byte(fmt.Sprintf(`{"spec":{"plugins":["test"]}}`))
+
+			_, err := h.Dynamic().Resource(schema.GroupVersionResource{
+				Group: "operator.openshift.io", Version: "v1",
+				Resource: "consoles",
+			}).Patch(ctx, "cluster", types.MergePatchType, patchData, metav1.PatchOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// revret the changes
+			patchEmpty := []byte(fmt.Sprintf(`{"spec":{"plugins":[""]}}`))
```

Comment:
I am not sure if patch it the second time to revert is necessary.
As the patched console plugin is dummy, it should not have problem to just leave it there?
Or if we would like to check the permission only, it should also be ok if we do not patch the `.spec`

Reply:
Thank you @bmeng I have removed the revert patch process as this is harmless to be patch with any dummy data.

Reply:
I'd generally recommend that tests should be leaving the cluster in the state that the cluster originally started in, and therefore it should be doing the removal. 

If a test is re-run on a cluster and the previous execution of the test has left the cluster in a different state, it's testing a different situation.